gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1582,8 +1582,13 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 							lhs0 := n.Lhs[0].(*NameExpr).Name
 							lhs1 := n.Lhs[1].(*NameExpr).Name
 
-							dt := evalStaticTypeOf(store, last, cx.X)
-							mt := baseOf(dt).(*MapType)
+							var mt *MapType
+							st := evalStaticTypeOf(store, last, cx.X)
+							if dt, ok := st.(*DeclaredType); ok {
+								mt = dt.Base.(*MapType)
```

Comment:
This will panic if the base of the declared type is not a map type.  I think this could be avoided by doing `bt := baseOf(st)` and then doing a type assertion on bt validate it is a map type.  Or even `bt := baseOf(evalStaticTypeOf(store, last, cx.X))`

Reply:
thank you for the comments. Agree, It does make sense to optimize it with better panic messages.